### PR TITLE
Add OpenApi Models

### PR DESCRIPTION
**Pull request by FatulM**

### CODE REVIEW

**Review by easazade — COMMENTED** [review]
Nice job

File: lib/model/header.dart [hunk 1]
```diff
@@ -0,0 +1,32 @@
+part of 'model.dart';
+
+class Header {
+  final bool? required;
```
[the remark this answers is not in the document]
Comment (payam-zahedi) [reply lib/model/header.dart:4]: I don't know?

File: lib/model/parameter.dart [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  final bool? required;
+
+  final bool? deprecated;
+
+  /// this parameter is going to depreciate in
+  /// the following versions of the Open Api Spec
+  final bool? allowEmptyValue;
+
+  final String? style;
+
```
[the remark this answers is not in the document]
Comment (easazade) [reply lib/model/parameter.dart]: Let's write in english from now on

File: lib/src/utils/utililty_functions.dart [hunk 3]
```diff
@@ -55,7 +55,7 @@ Future<Map<String, dynamic>> readJsonOrYamlFile(File file) async {
       return json;
     } else {
       YamlMap yaml = loadYaml(fileContent);
-      var map = yaml.map((key, value) => MapEntry(key.toString(), value));
+      var map = jsonDecode(jsonEncode(yaml)) as Map<String, dynamic>;
```
Comment (payam-zahedi) [inline lib/src/utils/utililty_functions.dart:58]: What was the problem with the above solution? 
is there any issue? 
I think using JSON encode and decode will increase time complexity.

File: test/openapi/model/pathitem/path_item_test.dart [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        // when we parse Response from map
+        var pathItem = PathItem.fromMap(map);
+        // then
+        assert(pathItem.post != null);
+        assert(pathItem.put != null);
+        assert(pathItem.get == null);
+        assert(pathItem.delete == null);
+        assert(pathItem.head == null);
+        assert(pathItem.options == null);
+        assert(pathItem.patch == null);
```
Comment (payam-zahedi) [inline test/openapi/model/pathitem/path_item_test.dart:25]: We decided to not using asserts.
Instead, you can write something like this.
``` dart
expect(pathItem.post , isNotNull);
```

You can also check the type:
``` dart
expect(pathItem.post , isA<Type>());
```

File: test/openapi/model/responses/responses_test.dart [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    test(
+      'should parse responses object from responses.yaml without any errors',
+      () async {
+        // with openapi responses defined in map
+        var map = await readJsonOrYamlFile(
+            File('test/openapi/model/responses/responses.yaml'));
+        // when we parse Responses from map
+        var responses = Responses.fromMap(map);
+        // then
+        assert(responses.defaultValue != null);
```
Comment (payam-zahedi) [inline test/openapi/model/responses/responses_test.dart:20]: We also have an `assert` here.

**Comment by FatulM** [comment]
Thanks
Some changes for schema may be applied

**Pull request merged**